### PR TITLE
make resources timeline optional

### LINKED ISSUE #8: Computation of resources timeline and addition of the header should be an option
Because not everyone needs that, we should make it configurable wether we want to compute those timings or not.

### PR DESCRIPTION
**Pull request by tuxnco**
Also, since there's now a bunch of args we pass to Phantom, let's pack all of them
into a general purpose args object, refered to as options within Phantom.

fixes #8

### CODE REVIEW

**Comment by panthony** [comment]
LGTM

Wonder why there is a part in a "settings" object and a part passed as environment variables

File: lib/server.js [hunk 1]
```diff
@@ -164,8 +172,7 @@ server.onPhantomPageCreate = function(req, res) {
     this.clearLocalStorage(req, res);
     this.clearMemoryCache(req, res);
 
-    req.prerender.page.run(_this.options.blockedResources || blockedResources, !!_this.options.logRequests, USER_AGENT, function(blockedResources, logRequests, ua, resolve, reject) {
-
+    req.prerender.page.run(args, function(options, resolve, reject) {
```
Comment (tuxnco) [inline lib/server.js:175]: @panthony within this anonymous function, variables defined out of the scope are _not_ available. The args and the anonymous function are serialized using json and sent to to Phantom JS, which in turns deserializes it run the serialized function with the serialized args.

File: lib/server.js [hunk 2]
```diff
@@ -571,8 +590,8 @@ server._sendResponse = function(req, res, options) {
         res.removeHeader('Content-Length');
     }
 
-    if (req.prerender.resourcesData) {
-        res.setHeader('X-OnCrawl-Resources-Timeline', JSON.stringify(req.prerender.resourcesData));
+    if (MONITOR_RESOURCES && req.prerender.resourcesData) {
```
Comment (tuxnco) [inline lib/server.js:593]: here the global variable MONITOR_RESOURCES is available.

File: lib/server.js (hunk 1 above)
Comment (panthony) [reply lib/server.js:175]: @tuxnco I was refering to `this.settings.userAgent` in the anonymous function. Why the rest is not part of "settings" too

**Comment by tuxnco** [comment]
Ok got it. These settings are internal to phridge / phantom.js, I did not
want to mix our options with phantom's settings.

**Pull request merged**